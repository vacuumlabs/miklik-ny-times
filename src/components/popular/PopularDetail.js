import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import CardHeader from '@material-ui/core/CardHeader'
import IconButton from '@material-ui/core/IconButton'
import Typography from '@material-ui/core/Typography'
import BackIcon from '@material-ui/icons/ArrowBack'
import PropTypes from 'prop-types'
import React from 'react'
import { ItemPropType } from '../../app/popular/popularTypes'

export default function PopularDetail({ item, onClose }) {
  return (
    <Card>
      <CardHeader
        avatar={
          <IconButton onClick={onClose} aria-label="Back">
            <BackIcon />
          </IconButton>
        }
        title={item.title}
        subheader={item.authors}
      />
      <CardContent>
        <Typography variant="body1" component="p">
          {item.abstract}
        </Typography>
      </CardContent>
    </Card>
  )
}

PopularDetail.propTypes = {
  item: ItemPropType.isRequired,
  onClose: PropTypes.func.isRequired,
}
