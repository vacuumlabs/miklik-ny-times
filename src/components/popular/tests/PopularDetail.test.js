import React from 'react'
import { render } from '@testing-library/react'
import PopularDetail from '../PopularDetail'
import { firstArticle } from '../../../tests/articleFixture'
import { commonSetup } from '../../../tests/testUtils'

describe('PopularDetail', () => {
  commonSetup()

  it('prints detail', () => {
    const { container } = render(
      <PopularDetail item={firstArticle} onClose={() => 0} />
    )
    expect(container).toMatchSnapshot()
  })
})
