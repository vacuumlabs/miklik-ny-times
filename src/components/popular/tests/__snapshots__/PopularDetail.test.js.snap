// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopularDetail prints detail 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
  >
    <div
      class="MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-avatar"
      >
        <button
          aria-label="Back"
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
        >
          Kentucky Crowd Cheers Valedictorian’s Trump Quote, Then Learns Obama Said It
        </span>
        <span
          class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          By LAURA M. HOLSON
        </span>
      </div>
    </div>
    <div
      class="MuiCardContent-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        A high school valedictorian quoted Trump. The crowd roared. Then: “Just kidding,” he said. “That was Barack Obama.”
      </p>
    </div>
  </div>
</div>
`;
